services/web/server/src/simcore_service_webserver/projects/_crud_api_read.py

```diff
@@ -58,19 +59,24 @@ async def list_projects(
     search: str | None,
     order_by: OrderBy,
     folder_id: FolderID | None,
+    workspace_id: WorkspaceID | None,
 ) -> tuple[list[ProjectDict], int]:
     app = request.app
-    settings = get_application_settings(app)
     db = ProjectDBAPI.get_from_app_context(app)
 
     user_available_services: list[dict] = await get_services_for_user_in_product(
         app, user_id, product_name, only_key_versions=True
     )
 
+    if workspace_id:
+        # Verify user access to the specified workspace; raise an error if access is denied
+        await get_workspace(
+            app, user_id=user_id, workspace_id=workspace_id, product_name=product_name
+        )
```

Comment:
I think we should have a ```check_workspace_access``` function no? this would spare the comments (actually same thing we do with get_project everywhere...)